includes/wikia/GlobalTitle.php

```diff
@@ -100,8 +100,21 @@ public static function newMainPage( $city_id = null ) {
 			throw new \Exception( 'Invalid $city_id.' );
 		}
 
-		$mainPageName = self::newFromText( 'Mainpage', NS_MEDIAWIKI, $city_id )->getContent();
-		$title = self::newFromText( $mainPageName, NS_MAIN, $city_id );
+		$mainPageTitle = self::newFromText( 'Mainpage', NS_MEDIAWIKI, $city_id );
+		$mainPageName = str_replace( ' ', '_', $mainPageTitle->getContent() );
+		$namespace = NS_MAIN;
+		// support for non-MAIN namespace pages, based on Title::secureAndSplit method
+		$prefixRegexp = "/^(.+?)_*:_*(.*)$/S";
+		if ( preg_match( $prefixRegexp, $mainPageName, $m ) ) {
+			$mainPageTitle->loadContLang();
+			$namespaceIndex = $mainPageTitle->mContLang->getNsIndex( $m[1] );
+			if ( $namespaceIndex !== false ) {
+				$namespace = $namespaceIndex;
+				$mainPageName = $m[2];
+			}
+		}
+
+		$title = self::newFromText( $mainPageName, $namespace, $city_id );
```

Comment:
Can we, instead of all this magic above, make an internal HTTP call to MediaWiki API to get the main page title?

Example: http://poznan.wikia.com/api.php?action=query&meta=siteinfo

```xml
<general mainpage="Wszystko o Poznaniu" base="http://poznan.wikia.com/wiki/Wszystko_o_Poznaniu" sitename="Poznańska Wiki" generator="MediaWiki 1.19.24" phpversion="7.0.7-4+deb.sury.org~trusty+1" phpsapi="apache2handler" dbtype="mysql" dbversion="5.6.24-72.2-log" case="first-letter" rights="CC-BY-SA" lang="pl" fallback8bitEncoding="iso-8859-2" writeapi="" timezone="UTC" timeoffset="0" articlepath="/wiki/$1" scriptpath="" script="/index.php" variantarticlepath="" server="http://poznan.wikia.com" wikiid="plpoznan" time="2017-09-04T11:34:35Z" misermode="">
```

Reply:
Actually, why do we use `GlobalTitle` when all we need is a local wiki mainpage that can be taken from `Title::newMainPage` ?

Reply:
Cause we need to run it out of wiki context (API request from discussions)

Reply:
I'd rather fix GlobalTitle::newMainPage method cause it's used in a few more places.

Reply:
That's fine.

But please use `Title::newMainPage` in this API, because `GlobalTitle` is not really needed (as we discussed it yesterday).